File: app/src/main/java/fi/tuska/jalkametri/activity/AboutActivity.java
package fi.tuska.jalkametri.activity;

import android.content.Intent;
import android.content.pm.PackageManager.NameNotFoundException;
import android.content.res.Resources;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import fi.tuska.jalkametri.R;
import fi.tuska.jalkametri.util.LogUtil;

/**
 * Activity for showing an about screen. The screen shows information about the
 * software.
 *
 * @author Tuukka Haapasalo
 */
public class AboutActivity extends JalkametriActivity implements GUIActivity {

    private static final String TAG = "AboutActivity";

    private TextView versionNumber;
    private TextView versionText;
    private TextView versionShort;

    public AboutActivity() {
        super(R.string.title_about, R.string.help_about);
        setShowDefaultHelpMenu(true);
    }

    /*
     * Standard activity functions --------------------------------------------
     */
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_about);

        versionNumber = (TextView) findViewById(R.id.version_number);
        versionText = (TextView) findViewById(R.id.version_info);
        versionShort = (TextView) findViewById(R.id.version_short);

        initComponents();

        updateUI();
    }

    @Override
    protected void onStart() {
        super.onStart();
        updateUI();
    }

    /*
     * Activity item initialization --------------------------------------------
     */
    private void initComponents() {
        setVersionNumber();
    }

    public void setVersionNumber() {
        try {
            String vNum = getPackageManager().getPackageInfo(getPackageName(),
                    0).versionName;
            versionNumber.setText(vNum);
        } catch (NameNotFoundException e) {
            versionNumber.setText("0.0");
        }
    }

    @Override
    public void updateUI() {
        Resources res = getResources();
        final boolean hasLicense = getPrefs().isLicensePurchased();
        // Set the version text to reflect the license purchasing status
        versionShort.setText(res
                .getString(hasLicense ? R.string.version_licensed
                        : R.string.version_free));
        versionText.setText(res
                .getString(hasLicense ? R.string.about_version_licensed
                        : R.string.about_version_free));
    }

    /*
     * Custom actions ----------------------------------------------------------
     */
    public void onOKPressed(View v) {
        // Dismiss this activity when OK is pressed
        onBackPressed();
    }

    public void purchaseLicense(View v) {
        // Purchase the license
        if (getPrefs().isLicensePurchased()) {
            LogUtil.i(TAG, "License already bought, not repurchasing");
            // Already bought
            updateUI();
        }
    }

    public void checkLicense(View v) {
        // Check for existing license
        if (getPrefs().isLicensePurchased()) {
            LogUtil.i(TAG, "License already bought, not checking");
            updateUI();
        }
    }

    /*
     * Return from activities --------------------------------------------------
     */
    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        updateUI();
    }
}

File: app/src/main/java/fi/tuska/jalkametri/activity/AboutActivity.kt
package fi.tuska.jalkametri.activity

import android.content.Intent
import android.content.pm.PackageManager.NameNotFoundException
import android.os.Bundle
import android.view.View
import android.widget.TextView
import fi.tuska.jalkametri.R

class AboutActivity : JalkametriActivity(R.string.title_about, R.string.help_about), GUIActivity {

    var viewModel: ViewModel? = null

    init {
        setShowDefaultHelpMenu(true)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_about)
        viewModel = ViewModel(this)
        updateUI()
    }

    override fun onStart() {
        super.onStart()
        updateUI()
    }

    // Dismiss this activity when OK is pressed
    fun onOKPressed(v: View) = onBackPressed()

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        super.onActivityResult(requestCode, resultCode, data)
        updateUI()
    }

    override fun updateUI() {
        viewModel?.updateUI()
    }

    class ViewModel(val activity: AboutActivity) {
        private val versionNumber: TextView = activity.findViewById(R.id.version_number) as TextView
        private val versionText: TextView = activity.findViewById(R.id.version_info) as TextView
        private val versionShort: TextView = activity.findViewById(R.id.version_short) as TextView

        init {
            initComponents()
        }

        private fun initComponents() {
            setVersionNumber()
        }

        private fun setVersionNumber() {
            versionNumber.text = try {
                activity.packageManager.getPackageInfo(activity.packageName, 0).versionName
            } catch (e: NameNotFoundException) {
                "0.0"
            }

        }

        fun updateUI() {
            val res = activity.resources
            val hasLicense = activity.prefs.isLicensePurchased
            // Set the version text to reflect the license purchasing status
            versionShort.text = res.getString(if (hasLicense) R.string.version_licensed
            else R.string.version_free)
            versionText.text = res.getString(if (hasLicense) R.string.about_version_licensed
            else R.string.about_version_free)
        }

    }

    companion object {
        private val TAG = "AboutActivity"
    }
}
